### PR TITLE
[Draft] ImportSVG - Set DPI automatically where possible

### PR DESCRIPTION
**Pull request by Syres916**
...otherwise request user input

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, please confirm the following:

- [ ] Branch rebased on latest master `git pull --rebase upstream master`
- [ ] Unit tests confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [ ] Commit message is [well-written](https://chris.beams.io/posts/git-commit/)
- [ ] Commit message includes `issue #<id>` or `fixes #<id>` where `<id>` is the [associated MantisBT](https://freecadweb.org/wiki/tracker#GitHub_and_MantisBT) issue id if one exists

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.19 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=34586).

---

### CODE REVIEW

File: src/Mod/Draft/importSVG.py [hunk 1]
```diff
@@ -264,44 +265,48 @@ def getsize(length,mode='discard',base=1):
...
+
+                    '%': 100 #arbitrarily chosen; has to depend on vieport size or (for filling patterns) on bounding box
+                    }
+        if mode == 'css96.0' or mode == 'css90.0':
+            topx={
+                    '' : 1.0, #default
+                    'px' : 1.0,
+                    'pt' : 1.25,
+                    'pc' : 15,
+                    'mm' : float(mode[-4])/25.4,
```
Comment (LemonAndroid) [inline src/Mod/Draft/importSVG.py]: Why floats? Aren't they extremely inaccurate in some situations and better suited for quick calculations, where accuracy is not important?

Comment (Syres916) [reply src/Mod/Draft/importSVG.py]: I've been checking the distances to 6 decimal places and I thought I was just replacing hard coded numbers with the same. What do you suggest I use instead, the Draft.precision function?

File: src/Mod/Draft/importSVG.py [hunk 2]
```diff
@@ -457,6 +463,41 @@ def startElement(self, name, attrs):
...
+                            ):
+                            self.svgdpi = 90.0
+                        if (
+                            int(InksFullverlst[0]) > 0
+                            ):
+                            self.svgdpi = 96.0
+                    if not 'inkscape:version' in data:
+                        msgBox = QtGui.QMessageBox()
+                        msgBox.setText(translate("ImportSVG","This SVG file does not appear to have been produced by Inkscape. If it does not contain absolute units then a DPI setting will be used."))
+                        msgBox.setInformativeText(translate("ImportSVG","Do you wish to use 96dpi? Clicking No will revert to the older standard 90dpi"))
```
Comment (luzpaz) [inline src/Mod/Draft/importSVG.py]: Instead of 'Clicking NO will...' I suggest: `Choosing 'No' will...`

File: src/Mod/Draft/importSVG.py [hunk 3]
```diff
@@ -457,6 +463,41 @@ def startElement(self, name, attrs):
...
+                        msgBox.setDefaultButton(QtGui.QMessageBox.No)
+                        ret = msgBox.exec_()
+                        if ret == QtGui.QMessageBox.Yes:
+                            self.svgdpi = 96.0
+                        else:
+                            self.svgdpi = 90.0
+                        if ret:
+                            FreeCAD.Console.PrintMessage("****** User specified "+str(self.svgdpi)+"dpi ******\n")
+                    if self.svgdpi == 1.0:
+                        FreeCAD.Console.PrintWarning("This SVG file ("+InksDocName+") has an unrecognised format which means the dpi could not be determined, importing with 96dpi\n")
```
Comment (luzpaz) [inline src/Mod/Draft/importSVG.py]: `...could not be determined; therefore importing with 96dpi\n")`

File: src/Mod/Draft/importSVG.py [hunk 4]
```diff
@@ -264,44 +265,48 @@ def getsize(length,mode='discard',base=1):
...
+                    'px' : 1.0,
+                    'pt' : 1.25,
+                    'pc' : 15,
+                    'mm' : float(mode[-4])/25.4,
+                    'cm' : float(mode[-4])/254.0,
+                    'in' : float(mode[-4]),
+                    'em':  15, #arbitrarily chosen; has to depend on font size
+                    'ex':  10, #arbitrarily chosen; has to depend on font size
+
+                    '%': 100 #arbitrarily chosen; has to depend on vieport size or (for filling patterns) on bounding box
```
Comment (luzpaz) [inline src/Mod/Draft/importSVG.py]: `s/vieport/viewport/`

File: src/Mod/Draft/importSVG.py [hunk 5]
```diff
@@ -264,44 +265,48 @@ def getsize(length,mode='discard',base=1):
...
+                    'px' : 25.4/float(mode[-4]),
+                    'pt' : 1.25*25.4/float(mode[-4]),
+                    'pc' : 15*25.4/float(mode[-4]),
+                    'mm' : 1.0,
+                    'cm' : 10.0,
+                    'in' : 25.4,
+                    'em':  15*2.54/float(mode[-4]), #arbitrarily chosen; has to depend on font size
+                    'ex': 10*2.54/float(mode[-4]), #arbitrarily chosen; has to depend on font size
+
+                    '%': 100 #arbitrarily chosen; has to depend on vieport size or (for filling patterns) on bounding box
```
Comment (luzpaz) [inline src/Mod/Draft/importSVG.py]: `s/vieport/viewport/`

**Comment by yorikvanhavre** [comment]
Merged, thanks!

**Pull request merged**